examples/chip-tool/commands/common/DeviceScanner.cpp

```diff
@@ -55,7 +55,7 @@ CHIP_ERROR DeviceScanner::Stop()
 
 void DeviceScanner::OnNodeDiscovered(const DiscoveredNodeData & nodeData)
 {
-    auto & commissionData = nodeData.nodeData;
+    auto & commissionData = nodeData.Get<CommissionNodeData>();
```

Comment:
For all these casts, what guarantees that the data inside nodeData is the "correct type" ? we should have some form of test (at least VerifyOrDie if not log&return) on the content of the data.